### PR TITLE
TST: Collect/Use arithmetic test fixtures

### PR DESCRIPTION
**Pull request by jbrockmendel**
Takes over from #22350 (which would have been a rebasing nightmare), addressing AFAICT all unaddressed comments from there.

All remaining fixtures in tests/arithmetic are collected in tests/arithmetic/conftest.py
Fixtures are given more descriptive names and docstrings as requested in #22350.

A handful of tests are altered to use fixtures; in a number of cases their `pytest.mark.parametrize` arguments were already equivalent to one of the fixtures.

A few new broken cases are identified, specifically numeric `Series` and `DataFrame` `__mul__` or `__rdiv__` with a) non-nanosecond `timedelta64` or b) `Tick`.  (I'll open an Issue for these)

Some tests got cleaned up nicely by using `tm.assert_equal` instead of parametrizing with `assert_func`

### CODE REVIEW

**Comment by jbrockmendel** [comment]
>In the file pandas/tests/arithmetic/test_numeric.py, following are the PEP8 issues :

>> Line 154:19: W504 line break after binary operator

setup.cfg has W503 ignored for this this warning.  Why does pep8speaks think its W504?

**Comment by jbrockmendel** [comment]
Traditional resourceWarning in circleCI

File: pandas/tests/arithmetic/conftest.py [hunk 1]
```diff
@@ -28,14 +28,32 @@ def zero(request):
...
-def idx(request):
+def numeric_idx(request):
+    """
+    Several types of numeric-dtypes Index objects
+    """
     return request.param
 
 
+@pytest.fixture
+def tdser():
```
Comment (jreback) [inline pandas/tests/arithmetic/conftest.py]: should conform / rename these at some point (e.g. have been using: timedelta_series) for something like this

File: pandas/tests/arithmetic/conftest.py [hunk 2]
```diff
@@ -28,14 +28,32 @@ def zero(request):
...
+# Vector Fixtures
+
 @pytest.fixture(params=[pd.Float64Index(np.arange(5, dtype='float64')),
                         pd.Int64Index(np.arange(5, dtype='int64')),
-                        pd.UInt64Index(np.arange(5, dtype='uint64'))],
+                        pd.UInt64Index(np.arange(5, dtype='uint64')),
+                        pd.RangeIndex(5)],
                 ids=lambda x: type(x).__name__)
-def idx(request):
+def numeric_idx(request):
```
Comment (jreback) [inline pandas/tests/arithmetic/conftest.py]: numeric_index (can be future PR)

File: pandas/tests/arithmetic/conftest.py [hunk 3]
```diff
@@ -28,14 +28,32 @@ def zero(request):
...
     return request.param
 
 
+@pytest.fixture
+def tdser():
+    """
+    Return a Series with dtype='timedelta64[ns]', including a NaT.
+    """
+    return pd.Series(['59 Days', '59 Days', 'NaT'], dtype='timedelta64[ns]')
+
```
Comment (jreback) [inline pandas/tests/arithmetic/conftest.py]: these 2 fixtures should be pushed up to pandas/conftest.py as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks. my comments are for a future PR FYI

**Pull request merged**